### PR TITLE
fix(reactstrap): correct Table `responsive` prop types

### LINKED ISSUE #45545: [@types/reactstrap] Table responsive should be type of string | boolean instead of boolean
From the original:

```
const propTypes = {
  ...,
  responsive: PropTypes.oneOfType([PropTypes.bool, PropTypes.string]),
};
```

### PR DESCRIPTION
**Pull request by peterblazejewicz**
- property type corrected
- test amended to use added property version

https://github.com/reactstrap/reactstrap/blob/master/src/Table.js#L15

Thanks!

/cc @nagiek

Fixes #45545

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Add or edit tests to reflect the change. (Run with `npm test`.)

### CODE REVIEW

**Comment by danger-public** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## reactstrap ([<kbd>unpkg</kbd>](https://unpkg.com/browse/reactstrap@latest/))
was missing the following properties:
1. DropdownContext
1. CustomFileInput
1. PopperContent
1. PopperTargetHelper
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 1cc8b10c39bc07ca823a0c95d60452b554239ea7
</p>

File: types/reactstrap/reactstrap-tests.tsx [hunk 1]
```diff
@@ -3261,7 +3261,7 @@ class Example98 extends React.Component {
 class Example99 extends React.Component {
   render() {
     return (
-      <Table responsive>
+      <Table responsive="md">
```
Comment (kyletsang) [inline types/reactstrap/reactstrap-tests.tsx:3264]: Can you add this as a separate test?  We want to test `responsive` as both boolean and string.

Comment (kyletsang) [reply types/reactstrap/reactstrap-tests.tsx:3264]: Scratch that, just noticed that Example100 is exactly the same as 99

**Review by kyletsang — APPROVED** [review]

**Comment by peterblazejewicz** [comment]
Ready to merge :kiss:

**Pull request merged**